src/main/clm_varpar.F90

```diff
@@ -104,16 +105,19 @@ subroutine clm_varpar_init()
     ! !LOCAL VARIABLES:
     !
     character(len=32) :: subname = 'clm_varpar_init'  ! subroutine name
+    integer :: actual_numcft     ! Actual number of crops
     !------------------------------------------------------------------------------
 
     ! Crop settings and consistency checks
 
-    if (use_crop) then
-       numpft      = mxpft   ! actual # of patches (without bare)
-       numcft      =  64     ! actual # of crops
+    ! slevis: I tried ".or. get_do_transient_crops()" & got unhelpful bld error
+    !         so added this fsurdat check instead
+    if (use_crop .or. index(fsurdat,'78pfts') > 0) then
```

Comment:
I am very uncomfortable with this change. We should never have logic based on the file name.

Reply:
Yes, absolutely. It should query the metadata for anything it needs to know.

Reply:
I looked for an alternative:

As far as I can tell, this piece of code executes before the model reads any surface or landuse data and before the model defines get_do_transient_crops. As a result, I believe that I do not have access to numcft or ncid or get_do_transient_crops.

This does not justify keeping my hack; however, I think our goal is to converge to one set of input data so that the if-then-else block will go away altogether. We will be left only with these two lines (not surrounded by ifs):
`numpft        = mxpft   ! actual # of patches (without bare)
 actual_numcft =  64     ! actual # of crops`

Let me know if you disagree and think that I should invest the time making this if-then-else more robust.

Reply:
@ekluzek 's suggestion: have a new routine like surfrdGetMaxVeg that gets the necessary dimension size. Call that before setting the various values here.

Reply:
Notes from our meeting on 10/1:

in clm_initializeMod.F90
after call control_init
enter call surfrd_get_maxveg  # similar to other subroutines in surfrd...
before call clm_varpar_init

Reply:
This now done.